ray_julia_jll/gen/artifacts.jl

```diff
@@ -0,0 +1,137 @@
+end
+
+function download_asset(asset_url)
+    run(`$(wget()) $asset_url -P $DIR`)
+    filename = replace(basename(asset_url), "%2B" => "+") # TODO: better way to parse URL in unicode?
+    return joinpath(DIR, filename)
+end
+
+if abspath(PROGRAM_FILE) == @__FILE__
+
+    repo_path = abspath(joinpath(@__DIR__, "..", ".."))
+    pkg_url = remote_url(repo_path)
+
+    jll_path = joinpath(repo_path, "ray_julia_jll")
+    artifacts_toml = joinpath(jll_path, "Artifacts.toml")
+
+    # Read Project.toml
+    jll_project_toml = joinpath(jll_path, "Project.toml")
+    jll_project = read_project(jll_project_toml)
+    jll_version = jll_project.version
+
+    wrappers_dir = joinpath(jll_path, "src", "wrappers")
+
+    artifacts_urls = get_release_asset_urls(jll_version)
+
+    for artifact_url in artifacts_urls
+
+        artifact_path = download_asset(artifact_url)
+
+        m = match(TRIPLET_REGEX, basename(artifact_path))
+        if isnothing(m)
+            throw(ArgumentError("Could not parse host triplet from $(basename(artifact_path))"))
+        end
+
+        platform_triplet = m[:triplet]
+        julia_version = m[:julia_version]
+        platform = parse(BinaryPlatforms.Platform, platform_triplet)
+
+        bind_artifact!(
+            artifacts_toml,
+            "ray_julia",
+            tree_hash_sha1(artifact_path);
+            platform=platform,
+            download_info=[(artifact_url, sha256sum(artifact_path))],
+            force=true
+        )
+
+        host_wrapper = joinpath(wrappers_dir, "$platform_triplet-julia_version+$julia_version.jl")
+        cp("wrapper.jl.tmp", host_wrapper; force=true)
+    end
+
+    @info "Committing and pushing changes to Artifacts.toml for $jll_version"
+
+    message = "Generate artifacts for v$(jll_version)"
+
+    # TODO: ghr and LibGit2 use different credential setups. Double check what BB does here.
+    Base.shred!(LibGit2.CredentialPayload()) do credentials
+        LibGit2.with(LibGit2.GitRepo(repo_path)) do repo
+
+            # TODO: This allows empty commits
+            for file in readdir(wrappers_dir)
+                filepath = joinpath("ray_julia_jll", "src", "wrappers", file)
+                LibGit2.add!(repo, filepath)
+            end
+            LibGit2.add!(repo, joinpath("ray_julia_jll", "Artifacts.toml"))
+            LibGit2.commit(repo, message)
+
+            # Same as "refs/heads/$branch" but fails if branch doesn't exist locally
+            # TODO: Ensure no other files are staged before committing
+            # TODO: Ensure no changes between HEAD~main except to Artifacts.toml
+            branch = LibGit2.branch(repo)
+            branch_ref = LibGit2.lookup_branch(repo, branch)
+            refspecs = [LibGit2.name(branch_ref)]
+
+            # TODO: Expecting users to have their branch up to date. Pushing outdated
+            # branches will fail like normal git CLI
+            LibGit2.push(repo; refspecs, credentials)
+        end
+    end
```

Comment:
With the revised workflow I'd avoid automatically pushing and maybe even skip adding and committing as that lets users validate before committing. What do you think?

Reply:
yeah I agree - it can be harder to spot (and then reverse) unintended changes, which will probably lead to more commits and/or force pushes

Reply:
I've removed this block now